File: Interpreted/tak-function/tak.lua
function tak(x, y, z)
    return y<x and tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y)) or z
end

print(tak(tonumber(arg[1]), tonumber(arg[2]), tonumber(arg[3])));

File: Interpreted/tak-function/tak.rb
def tak(x, y, z)
    y < x ? tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y))
          : z;
end

puts tak(ARGV[0].to_i, ARGV[1].to_i, ARGV[2].to_i);
